third_party/move/move-compiler-v2/src/env_pipeline/inliner.rs

```diff
@@ -953,6 +1109,93 @@ impl<'env, 'rewriter> InlinedRewriter<'env, 'rewriter> {
             Some(new_args_expr)
         };
 
+        // Rewrite and instrument spec block for lambda
+        if let Some(spec) = spec_opt {
+            if let ExpData::SpecBlock(_, spec) = spec.as_ref() {
+                let new_body_node = || env.new_node(body_loc.clone(), body_type.clone());
+
+                // generate pattern for lambda body
+                let body_ty_vec = if let Type::Tuple(tys) = body_type.clone() {
+                    tys.clone()
+                } else {
+                    vec![body_type.clone()]
+                };
+                let mut body_pat_vec = vec![];
+                let mut ret_vars = vec![];
```

Comment:
You should document (in a comment) essentially would Move code you are generating here and elswhere.

Reply:
Added the comment after pulling out into a separate function.